include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -295,15 +295,17 @@ struct cartesian_segments
     <
         typename UniqueSubRange1,
         typename UniqueSubRange2,
-        typename Policy
+        typename Policy,
+        typename SideStrategy
     >
     static inline typename Policy::return_type
         apply(UniqueSubRange1 const& range_p,
               UniqueSubRange2 const& range_q,
-              Policy const& policy)
+              Policy const& policy,
+              SideStrategy const& side)
```

Comment:
Is the intention to allow to pass various side strategies in the same algorithm? Or is this changed to allow defining side strategy in one place only, i.e. in umbrella strategy and pass it into all strategies using it? If it's the former then this can lead to inconsistencies. If it's the latter then I don't understand why isn't it done for other coordinate systems too?

If the purpose is to "play" with the cartesian strategy then why not passing the side strategy as class template argument? This way the responsibility for the consistency of the strategies will rest only on the umbrella strategy.